cli/cli.go

```diff
@@ -0,0 +1,176 @@
+package cli
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/exercism/cli/debug"
+	update "github.com/inconshreveable/go-update"
+)
+
+var (
+	// BuildOS is the operating system (GOOS) used during the build process.
+	BuildOS string
+	// BuildARM is the ARM version (GOARM) used during the build process.
+	BuildARM string
+	// BuildARCH is the architecture (GOARCH) used during the build process.
+	BuildARCH string
+)
+
+var (
+	osMap = map[string]string{
+		"darwin":  "mac",
+		"linux":   "linux",
+		"windows": "windows",
+	}
+
+	archMap = map[string]string{
+		"amd64": "64bit",
+		"386":   "32bit",
+		"arm":   "arm",
+	}
+)
+
+var (
+	HTTPClient       = &http.Client{Timeout: 10 * time.Second}
+	LatestReleaseURL = "https://api.github.com/repos/exercism/cli/releases/latest"
+)
+
+type CLI struct {
+	Version       string
+	LatestRelease *Release
+}
+
+func New(version string) *CLI {
+	return &CLI{
+		Version: version,
+	}
+}
+
+func (c *CLI) IsUpgradeNeeded() (bool, error) {
```

Comment:
I got rid of the `upgrader` type, and now the `CLI` type can check whether or not it needs an upgrade.